cfme/generic_objects/definition/ui.py

```diff
@@ -0,0 +1,477 @@
+from . import GenericObjectDefinition, GenericObjectDefinitionCollection
+
+
+class GenericObjectDefinitionToolbar(View):
+    configuration = Dropdown(text='Configuration')
+    download = Dropdown(text='Download')
+
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class GenericObjectDefinitionView(BaseLoggedInPage):
+
+    @property
+    def in_generic_object_definition(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Automation', 'Automate']
+        )
+
+
+class AccordionForm(View):
+
+    @View.nested
+    class classes(Accordion):   # noqa
+        ACCORDION_NAME = "Generic Object Classes"
+        tree = ManageIQTree()
+
+
+class GenericObjectDefinitionAllView(GenericObjectDefinitionView):
+    toolbar = View.nested(GenericObjectDefinitionToolbar)
+    including_entities = View.include(BaseEntitiesView, use_parent=True)
+    accordion = View.nested(AccordionForm)
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_generic_object_definition and
+            self.toolbar.configuration.is_displayed and
+            self.entities.title.text == 'All Generic Object Classes'
+        )
+
+
+class ParametersForm(View):
+    ROOT = ParametrizedLocator('//generic-object-table-component[@key-type="{@param_type}"]')
+    ALL_PARAMETERS = './/input[contains(@class, "ng-not-empty")]'
+    add = Button(ParametrizedString('Add {@param_type}'))
+    name = Input(locator='.//input[contains(@class, "ng-empty")]')
+    type_class = BootstrapSelect(
+        locator='.//input[contains(@class, "ng-empty")]//ancestor::tr//button')
+
+    def __init__(self, parent, param_type, logger=None):
+        View.__init__(self, parent, logger=logger)
+        self.param_type = param_type
+
+    def all(self):
+        return [(element.get_attribute('value'), element.get_attribute('name'))
+                for element in self.browser.elements(self.ALL_PARAMETERS)]
+
+    @property
+    def empty_field_is_present(self):
+        try:
+            return self.browser.element(self.name)
+        except NoSuchElementException:
+            return False
+
+    def add_parameter_row(self):
+        if not self.empty_field_is_present:
+            self.add.click()
+
+    def fill(self, parameters):
+        if parameters:
+            if isinstance(parameters, dict):
+                for name, type_class in parameters.items():
+                    self.add_parameter_row()
+                    self.type_class.fill(type_class)
+                    self.name.fill(name)
+            elif isinstance(parameters, list):
+                for name in parameters:
+                    self.add_parameter_row()
+                    self.name.fill(name)
+
+    def delete(self, name):
+        all_params = self.all()
+        for param in all_params:
+            param_name, element_name = param
```

Comment:
This can be unpacked within the for loop:
`for param_name, element_name in all_params`